src/target/source/codegen_metal.cc

```diff
@@ -47,7 +47,7 @@ CodeGenMetal::CodeGenMetal() {
   decl_stream << "#include <metal_stdlib>\n";
   decl_stream << "using namespace metal;\n\n";
   decl_stream << "union __TVMArgUnion {\n"
-              << " int v_int;\n"
+              << " int v_long;\n"
```

Comment:
should it be int v_int[2]?

Reply:
if I add bits()==64 support explicitly (which doesn't use ArgUnion), is there any case we will use `ArgUnion`? Looks like previous this union hardcoded to only support `int`

Reply:
previously it was intended for passing in lower number of bits (e.g. uint8 and i16). But not sure the driver side support it yet

Reply:
https://github.com/apache/tvm/blob/7d1f47f8ac49718f05981ee89c56f18d7bfb4d01/src/target/source/codegen_metal.cc#L113 Variables in this union is referred by type e.g `v_int, v_char`, but the `ArgUnion` only had `v_int` so I this it won't work.
I can keep `v_int[2]` for now

Reply:
sounds good. I think that part also needs some work then